rosbag2_cpp/src/rosbag2_cpp/reindexer.cpp

```diff
@@ -189,6 +189,7 @@ void Reindexer::aggregate_metadata(
     rosbag2_cpp::ConverterOptions blank_converter_options {};
     bag_reader->open(temp_so, blank_converter_options);
     auto temp_metadata = bag_reader->get_metadata();
+    metadata_.storage_identifier = temp_metadata.storage_identifier;
```

Comment:
Note: before this was leaving the storage ID blank in a reindex operation, if storage id not specified by user.